modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -249,11 +228,11 @@ def _define_index(
         row_lengths : list
             Partitions rows lengths.
         """
-        if index_col is None:
-            row_lengths = cls.materialize(index_ids)
-            new_index = pandas.RangeIndex(sum(row_lengths))
+        index_objs = cls.materialize(index_ids)
+        if len(index_objs) == 0 or isinstance(index_objs[0], int):
```

Comment:
Is condition `if len(index_objs) == 0 or isinstance(index_objs[0], int):` exactly equivalent to condition `if index_col is None:`?

Reply:
I hope so, i can't imagine scenario where these conditions are not equivalent.
In the parsers.py if `index_col is None` we will obtain range index as a result of `pandas.read_csv` that will be replaced with integer.

Reply:
@amyskov I am thinking that there may be new cases in which numeric Series is replaced by range Series.

Reply:
Can't catch your idea, please provide some example when we use `read_csv` with `index_col=None` and obtain non-range index...
This line just checks if workers returned integers for index (this occurs only if resulting indexes on the workers are range typed), and if so to replace it with RangeIndex.

Reply:
It turns out that I did not pay attention to that `index_objs` can contain several objects of the type Series. No more questions.